rules.py

```diff
@@ -826,7 +831,40 @@ def check(self):
         if error_log:
             raise base.ElectionTreeError(
                 "The Election File contains unreferenced Candidates", error_log)
-                
+
+
+class DuplicateContestNames(base.TreeRule):
+    """Check that the file contains unique ContestNames.
+        Add Warning if duplicate ContestName found."""
+
+    def check(self):
+        name_contest_id = {}  # Mapping for <Name> and it's Contest ObjectId.
+        error_log = []
+        for event, element in etree.iterwalk(self.election_tree):
+            tag = self.strip_schema_ns(element)
+            if tag != 'Contest':
```

Comment:
Nit: Use " instead of ' as quote marks here since that's how the rest of the file does it.